hudi-utilities/src/test/java/org/apache/hudi/utilities/sources/TestS3EventsHoodieIncrSource.java

```diff
@@ -470,18 +472,23 @@ public void testSplitSnapshotLoad(String snapshotCheckPoint, String exptected1,
     List<Triple<String, Long, String>> filePathSizeAndCommitTime = new ArrayList<>();
     // Add file paths and sizes to the list
     filePathSizeAndCommitTime.add(Triple.of("path/to/file1.json", 50L, "1"));
+    filePathSizeAndCommitTime.add(Triple.of("path/to/file_no_match1.json", 50L, "1"));
     filePathSizeAndCommitTime.add(Triple.of("path/to/file2.json", 50L, "1"));
     filePathSizeAndCommitTime.add(Triple.of("path/to/skip1.json", 50L, "2"));
     filePathSizeAndCommitTime.add(Triple.of("path/to/skip2.json", 50L, "2"));
+    filePathSizeAndCommitTime.add(Triple.of("path/to/file_no_match2.json", 50L, "2"));
     filePathSizeAndCommitTime.add(Triple.of("path/to/file5.json", 50L, "3"));
     filePathSizeAndCommitTime.add(Triple.of("path/to/file4.json", 50L, "3"));
+    filePathSizeAndCommitTime.add(Triple.of("path/to/file_no_match3.json", 50L, "3"));
 
     Dataset<Row> inputDs = generateDataset(filePathSizeAndCommitTime);
 
     setMockQueryRunner(inputDs, Option.of(snapshotCheckPoint));
     when(mockCloudObjectsSelectorCommon.loadAsDataset(Mockito.any(), Mockito.any(), Mockito.any(), Mockito.eq(schemaProvider), Mockito.anyInt())).thenReturn(Option.empty());
     TypedProperties typedProperties = setProps(READ_UPTO_LATEST_COMMIT);
     typedProperties.setProperty("hoodie.streamer.source.s3incr.ignore.key.prefix", "path/to/skip");
+    typedProperties.setProperty("hoodie.streamer.source.cloud.data.select.path.regex", "path/to/file[0-9]+");
```

Comment:
Could you add other test cases on how this new config works together with `source.cloud.data.select.relpath.prefix` if set?

Reply:
Have added them below ?

Reply:
The tests look OK to me now.